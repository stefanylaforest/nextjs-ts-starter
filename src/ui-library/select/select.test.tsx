import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Select from './select';

describe('Select input', () => {
  it('should have a placeholder option', () => {
    render(
      <Select
        placeholder="choose a number"
        name={'choose one'}
        options={['1', '2', '3']}
        onChange={jest.fn()}
      />
    );
    expect(screen.getByRole('option', { name: 'choose a number' })).toHaveValue('');
  });

  it('should select an option', async () => {
    render(<Select name={'choose one'} options={['1', '2', '3']} onChange={jest.fn()} />);
    await userEvent.selectOptions(screen.getByRole('combobox'), '3');
    expect((screen.getByRole('option', { name: '1' }) as HTMLOptionElement).selected).toBe(false);
    expect((screen.getByRole('option', { name: '2' }) as HTMLOptionElement).selected).toBe(false);
    expect((screen.getByRole('option', { name: '3' }) as HTMLOptionElement).selected).toBe(true);
  });

  it('should have a default value', () => {
    render(
      <Select
        name={'choose one'}
        options={['a', 'b', 'c']}
        onChange={jest.fn()}
        defaultValue={'b'}
      />
    );
    expect((screen.getByRole('option', { name: 'a' }) as HTMLOptionElement).selected).toBe(false);
    expect((screen.getByRole('option', { name: 'b' }) as HTMLOptionElement).selected).toBe(true);
    expect((screen.getByRole('option', { name: 'c' }) as HTMLOptionElement).selected).toBe(false);
  });
});
