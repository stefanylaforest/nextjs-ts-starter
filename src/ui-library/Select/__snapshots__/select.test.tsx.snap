// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Select Input should match snapshot 1`] = `
<div>
  <div>
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="arrowSelect"
      data-testid="listbox-button"
      role="button"
      tabindex="0"
    >
      <span />
      <svg
        class="h-6 w-6"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m19 9-7 7-7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
</div>
`;
